setup.py

```diff
@@ -1,134 +1,23 @@
-#!/usr/bin/env python
+import subprocess
 
-from wagtail import __version__
-from wagtail.utils.setup import assets, check_bdist_egg, sdist
+from setuptools import setup
+from setuptools.command.sdist import sdist as base_sdist
 
-try:
-    from setuptools import find_packages, setup
-except ImportError:
-    from distutils.core import setup
 
+class sdist(base_sdist):
+    def compile_assets(self):
+        try:
+            subprocess.check_call(["npm", "run", "build"])
+        except (OSError, subprocess.CalledProcessError) as e:
+            print("Error compiling assets: " + str(e))  # noqa: T201
+            raise SystemExit(1)
```

Comment:
Moving this from `wagtail.utils.setup` to here because with `build` as the frontend, it doesn't allow imports from wagtail itself during the build without installing it in the build environment

https://github.com/pypa/build/issues/572